sql/analyzer/scope.go

```diff
@@ -80,6 +79,8 @@ func (s *Scope) newScopeFromSubqueryAlias(sqa *plan.SubqueryAlias) *Scope {
 		// We don't include the current inner node so that the outer scope nodes are still present, but not the lateral nodes
 		if s.currentNodeIsFromSubqueryExpression {
 			sqa.OuterScopeVisibility = true
+		}
+		if sqa.OuterScopeVisibility {
```

Comment:
this is the change that concerns me the most, can't explain yet

Reply:
my later changes appeared to resolved this, still don't remember which test broke, but reverting back to the original code